skbuild/resources/cmake/FindPythonExtensions.cmake

```diff
@@ -317,6 +317,31 @@ if(NOT DEFINED PYTHON_EXTENSION_MODULE_SUFFIX)
   set(PYTHON_EXTENSION_MODULE_SUFFIX "${_item}")
 endif()
 
+function(_set_python_extension_symbol_visibility _target)
```

Comment:
To support both python2 and python3, I suggest to have something like this:

```
function(_set_python_extension_symbol_visibility _target)
[...]
  if(PYTHON_VERSION_MAJOR VERSION_GREATER 2)
    set(module_init_function_prefix "PyInit_")
  else()
    set(module_init_function_prefix "init")
  endif()

  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
    set_target_properties(${_target} PROPERTIES LINK_FLAGS 
        "/EXPORT:${module_init_function_prefix}${_target}"
    )
  elseif("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
    set(_script_path
      ${CMAKE_CURRENT_BINARY_DIR}/_skbuild/${_target}-version-script.map
    )
    file(WRITE
      ${_script_path}
      "
      {
        global:
          ${module_init_function_prefix}${_target};
        local: *;
      }
      "      
    )
    set_target_properties(${_target} PROPERTIES LINK_FLAGS 
        "-Wl,--version-script=${_script_path}"
    )
  endif()
endfunction()
```

Reply:
Also, instead of `_skbuild`, we should introduce the equivalent of `SKBUILD_DIR` (already available in python as `skbuild.constant.SKBUILD_DIR`) in CMake

Reply:
Since extension module includes `PyMODINIT_FUNC`, the associated symbol are already exported. For example, see [here](https://github.com/scikit-build/scikit-build/blob/3f5cfc659e4270f50a55d6e0b0bee0561e9c520b/tests/samples/hello-cpp/hello/_hello.cxx#L61)

On linux, using `readelf -s /path/to/module.so` also confirmed that the symbol is already global.

I suspect that visibility need to be set only for the cython generated modules.

Reply:
The purpose of the script on Linux to hide all of the other symbols. Can you confirm the script has the intended effect?